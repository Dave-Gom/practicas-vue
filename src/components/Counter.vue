<template>
  <div>
    <h2>{{ title || "Counter" }}</h2>
    <p>{{ counter }}<sup>2</sup> = {{ squareValue }}</p>

    <div class="buttons">
      <button v-on:click="addOne">+1</button>
      <button v-on:click="minusOne">-1</button>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    title: String,
    start: {
      type: Number,
      required: true,
      validator(value) {
        return value > 100;
      },
    },
  },
  data() {
    return { counter: this.start };
  },
  methods: {
    getSquareValue() {
      return "hola Mundo";
    },
    addOne() {
      this.counter = this.counter + 1;
    },
    minusOne() {
      this.counter = this.counter - 1;
    },
  },
  computed: {
    squareValue() {
      return this.counter * this.counter;
    },
  },
};
</script>

<style>
button {
  background-color: #64b687;
  border-radius: 5px;
  border: 0.4px solid white;
  color: white;
  cursor: pointer;
  margin: 0 5px;
  padding: 5px 15px;
  transition: 0.3s ease-in-out;
}
</style>
