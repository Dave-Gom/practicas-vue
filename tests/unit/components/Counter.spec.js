import Counter from "../../../src/components/Counter";
import { shallowMount, mount } from "@vue/test-utils";

describe("Counter Component", () => {
  /* test("debe de sei igual al snapshot ", () => {
    const wrapper = shallowMount(Counter);
    expect(wrapper.html()).toMatchSnapshot();
  }); */

  test("debe tener el valor por defecto", () => {
    const wrapper = shallowMount(Counter);
    const h2 = wrapper.find("h2");
    console.log(h2);
    expect(h2.text()).toBe("Counter");
  });
});
