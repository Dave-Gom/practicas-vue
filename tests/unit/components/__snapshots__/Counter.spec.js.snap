// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter debe de sei igual al snapshot  1`] = `
<div>
  <h2>Counter</h2>
  <p><sup>2</sup> = NaN</p>
  <div><button>+1</button><button>-1</button></div>
</div>
`;
